src/main/java/org/folio/oaipmh/helpers/AbstractHelper.java

```diff
@@ -1,32 +1,58 @@
 package org.folio.oaipmh.helpers;
 
+import io.vertx.core.json.JsonObject;
+import org.folio.oaipmh.Request;
+import org.folio.okapi.common.XOkapiHeaders;
+import org.folio.rest.tools.client.HttpClientFactory;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+import org.folio.rest.tools.utils.TenantTool;
+import org.openarchives.oai._2.HeaderType;
 import org.openarchives.oai._2.OAIPMH;
-import org.openarchives.oai._2.ObjectFactory;
-import org.openarchives.oai._2.VerbType;
+import org.openarchives.oai._2.RequestType;
+import org.openarchives.oai._2_0.oai_identifier.OaiIdentifier;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
+import java.util.Map;
 
 /**
  * Abstract helper implementation that provides some common methods.
  */
 public abstract class AbstractHelper implements VerbHelper {
+
   public static final String REPOSITORY_BASE_URL = "repository.baseURL";
 
-  protected ObjectFactory objectFactory = new ObjectFactory();
+  /**
+   * The value should be populated only once at start up. So considered to be constant.
+   * @see #createRepositoryIdentifierPrefix()
+   */
+  private static String OAI_IDENTIFIER_PREFIX;
+
+  /**
+   * Holds instance to handle items returned
+   */
+  protected static ItemsStorageHelper storageHelper;
+
+  /**
+   * Initializes data required for OAI-PMH responses
+   */
+  public static void init() {
```

Comment:
I'm not sure if it's a good idea to introduce this method and make InitAPI to call it. Specifically because instances of the helpers which extend this AbstractHelper are created in static block, i.e. earlier than init() method is invoked.

One of the options is to initialize OAI_IDENTIFIER_PREFIX lazily in createRepositoryIdentifierPrefix():
  private String createRepositoryIdentifierPrefix() {
     if (OAI_IDENTIFIER_PREFIX == null) {
        try{
        // init OAI_IDENTIFIER_PREFIX
        return OAI_IDENTIFIER_PREFIX;
        } catch (MalformedURLException e) {
           throw new IllegalStateException(e);
        }
    } else {
       return OAI_IDENTIFIER_PREFIX;
    }
  }

Even though it might happen that several threads will try to initialize the prefix, it's not a problem since all of them will produce the same value.
The benefit is that when someone needs the prefix, it is guaranteed to be available.

@craigmcnally , @piotr-kalashuk 
what do you think?

Reply:
How about creating a static method to instantiate the helpers which is also called from InitAPI rather than doing so in a static block?

Reply:
@craigmcnally we will introduce static method in next PR